packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -194,6 +196,10 @@ export default class InternalModel {
     this._isDematerializing = false;
     this._scheduledDestroy = null;
 
+    this._record = null;
```

Comment:
Are these for shape?

Reply:
yes. We previously relied on calling `reset` to set them initially but we no longer want to do so because of the change to `currentState`